orttraining/orttraining/core/framework/ortmodule_graph_builder.cc

```diff
@@ -206,6 +210,21 @@ Status OrtModuleGraphBuilder::BuildGradientGraph(const std::unordered_set<std::s
   return Status::OK();
 }
 
+void OrtModuleGraphBuilder::GetFrontierTensors() {
+  const Graph& graph = gradient_model_->MainGraph();
+  for (const auto& param : graph_info_.initializer_names_to_train) {
+    std::vector<const Node*> consumer_nodes = graph.GetConsumerNodes(param);
+    // Initial support is limited to caching Cast output. This can
+    // be extended to accomodate more ops whose result depends only
+    // on the weight tensor which is a WIP.
+    for (const Node* node : consumer_nodes) {
+      if (node != nullptr && node->OpType() == "Cast") {
+        graph_info_.frontier_node_arg_map[param] = node->OutputDefs()[0]->Name();
```

Comment:
>graph_info_.frontier_node_arg_map[param] = node->OutputDefs()[0]->Name(); [](http://example.com/codeflow?start=8&length=73)

nit: might be good to assert that "cast" op has exactly one output.

Reply:
it does not need to have just one output